### PR TITLE
fix: template merging for bind mounts [DET-4630]

### PR DESCRIPTION
**Pull request by shiyuann**
## Description

Fix merging template for bind mounts.

## Test Plan

Manually tested it.

## Commentary (optional)

N/A

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

File: master/pkg/model/experiment_config.go [hunk 1]
```diff
@@ -250,6 +250,19 @@ func (r OptimizationsConfig) Validate() []error {
 	}
 }
 
+// BindMountsConfig is the configuration for bind mounts.
+type BindMountsConfig []BindMount
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (b *BindMountsConfig) UnmarshalJSON(data []byte) error {
+	unmarshaled := make([]BindMount, 0)
```
Comment (dzhu) [inline master/pkg/model/experiment_config.go:258]: non-blocking: I like `var unmarshaled []BindMount` a little better (something about not distinguishing empty slices from `nil` ones), though to be honest I'm still not sure what would be considered better Go style.

File: master/pkg/model/experiment_config.go [hunk 2]
```diff
@@ -250,6 +250,19 @@ func (r OptimizationsConfig) Validate() []error {
...
+	if err := json.Unmarshal(data, &unmarshaled); err != nil {
+		return errors.Wrapf(err, "failed to parse bind mounts")
```
Comment (dzhu) [inline master/pkg/model/experiment_config.go:260]: non-blocking: No need for formatting here:
```suggestion
		return errors.Wrap(err, "failed to parse bind mounts")
```

**Review by dzhu — APPROVED** [review]

File: master/pkg/model/experiment_config.go (hunk 1 above)
Comment (shiyuann) [reply master/pkg/model/experiment_config.go:258]: Why do you want not to distinguish empty slices from nil ones? Golang doesn't always prevent nil dereference.

Comment (dzhu) [reply master/pkg/model/experiment_config.go:258]: Oh wait, there actually is a definitive answer on this: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices. (Well, that distinguishes `var t []T` from `t := []T{}`, but all the points apply the same if the latter is replaced by `make([]T, 0)`. Nil slices can't be dereferenced like nil pointers can, so I don't think dereferencing is relevant here.

**Pull request merged**